### ISSUE TITLE
XCode 11 warns about 'aligned_alloc' availability

### ISSUE THREAD

**Issue by Valakor** [GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator#79]
I just upgraded to XCode 11 and it looks like Apple has finally implemented `aligned_alloc` as per the C++17 spec. When compiling (I'm targeting 10.12) I now get a warning:
`'aligned_alloc' is only available on macOS 10.15 or newer`.

I assume the definition of aligned_alloc in vk_mem_alloc.h doesn't cause any redefinition errors because Apple's implementation doesn't exist when I'm targeting 10.12, but it's likely there would be problems if I was targeting 10.15 or greater. Unfortunately, I'm unfamiliar with the macros required to check the targeted OS versions so I can't offer a pull request right now.

Snippet from _malloc.h for the curious (all credit to Apple, etc. etc.)
```c++
#if (__DARWIN_C_LEVEL >= __DARWIN_C_FULL) && \
        ((defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L) || \
         (defined(__cplusplus) && __cplusplus >= 201703L))
void    *aligned_alloc(size_t __alignment, size_t __size) __result_use_check __alloc_size(2) __OSX_AVAILABLE(10.15) __IOS_AVAILABLE(13.0) __TVOS_AVAILABLE(13.0) __WATCHOS_AVAILABLE(6.0);
#endif
```

**Comment by adam-sawicki-amd** [comment]
Thank you for reporting this issue. Unfortunately I don't have Apple platform to test this. My primary platform is PC Windows. I will leave this issue ticket open in case someone could provide a solution.

**Comment by Kristine1975** [comment]
The version checks are done using the `availability` attribute: http://clang.llvm.org/docs/AttributeReference.html#availability

But I think this has been fixed with #137.

**Comment by adam-sawicki-a** [comment]
I'm closing this ticket due to inactivity. Please let me know if you have additional questions.
